src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -136,6 +134,8 @@ private void TrackPush2(int i1, int i2)
 
         private void Backtrack()
         {
+            CheckTimeout();
```

Comment:
For Interpreter do we also need to do some checks for LookArounds?

Reply:
Good catch.